src/OlStyleParser.ts

```diff
@@ -763,6 +760,17 @@ export class OlStyleParser implements StyleParser<OlStyleLike> {
           rule.symbolizers.forEach((symb: Symbolizer) => {
             const olSymbolizer: any = this.getOlSymbolizerFromSymbolizer(symb, feature);
 
+            if (symb.visibility === false) {
+              return;
+            }
+
+            if (isGeoStylerBooleanFunction(symb.visibility)) {
+              const visibility = OlStyleUtil.evaluateBooleanFunction(symb.visibility);
+              if (!visibility) {
+                return;
+              }
+            }
+
```

Comment:
The diff of this suggestion looks a bit odd but it only adds `null` to the return value as setting the style to `undefined` will result in the default style. It also moves `getOlSymbolizerFromSymbolizer` behind the checks for performance reasons.

```suggestion
            if (symb.visibility === false) {
              return null;
            }

            if (isGeoStylerBooleanFunction(symb.visibility)) {
              const visibility = OlStyleUtil.evaluateBooleanFunction(symb.visibility);
              if (!visibility) {
                return null;
              }
            }

            const olSymbolizer: any = this.getOlSymbolizerFromSymbolizer(symb, feature);
```